lib/WiFi/src/WiFiEventHandler.cpp

```diff
@@ -0,0 +1,268 @@
+    return ESP_OK;
+} // wifiReady
+
+
+esp_err_t WiFiEventHandler::staStart()
+{
+    ESP_LOGD(LOG_TAG, "default staStart");
+
+    return ESP_OK;
+} // staStart
+
+
+esp_err_t WiFiEventHandler::staStop()
+{
+    ESP_LOGD(LOG_TAG, "default staStop");
+    return ESP_OK;
+} // staStop
+
+
+/**
+ * @brief Handle the Station Connected event.
+ * Handle having connected to remote AP.
+ * @param [in] event_connected system_event_sta_connected_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::staConnected(system_event_sta_connected_t info)
+{
+    ESP_LOGD(LOG_TAG, "default staConnected");
+    return ESP_OK;
+} // staConnected
+
+
+/**
+ * @brief Handle the Station Disconnected event.
+ * Handle having disconnected from remote AP.
+ * @param [in] event_disconnected system_event_sta_disconnected_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::staDisconnected(system_event_sta_disconnected_t info)
+{
+    ESP_LOGD(LOG_TAG, "default staDisconnected");
+    return ESP_OK;
+} // staDisconnected
+
+
+/**
+ * @brief Handle a Station Connected to AP event.
+ * Handle having a station connected to ESP32 soft-AP.
+ * @param [in] event_sta_connected system_event_ap_staconnected_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::apStaConnected(system_event_ap_staconnected_t info)
+{
+    ESP_LOGD(LOG_TAG, "default apStaConnected");
+    return ESP_OK;
+} // apStaConnected
+
+
+/**
+ * @brief Handle a Station Disconnected from AP event.
+ * Handle having a station disconnected from ESP32 soft-AP.
+ * @param [in] event_sta_disconnected system_event_ap_stadisconnected_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::apStaDisconnected(system_event_ap_stadisconnected_t info)
+{
+    ESP_LOGD(LOG_TAG, "default apStaDisconnected");
+    return ESP_OK;
+} // apStaDisconnected
+
+
+/**
+ * @brief Handle a Scan for APs done event.
+ * Handle having an ESP32 station scan (APs) done.
+ * @param [in] event_scan_done system_event_sta_scan_done_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::staScanDone(system_event_sta_scan_done_t info)
+{
+    ESP_LOGD(LOG_TAG, "default staScanDone");
+    return ESP_OK;
+} // staScanDone
+
+
+/**
+ * @brief Handle the auth mode of APs change event.
+ * Handle having the auth mode of AP ESP32 station connected to changed.
+ * @param [in] event_auth_change system_event_sta_authmode_change_t.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::staAuthChange(system_event_sta_authmode_change_t info)
+{
+    ESP_LOGD(LOG_TAG, "default staAuthChange");
+    return ESP_OK;
+} // staAuthChange
+
+
+WiFiEventHandler::~WiFiEventHandler()
```

Comment:
Could this be removed? we don't use it currently

Reply:
do you mean the destructor?

Reply:
yes that one